### PR TITLE
TimelineBody height fix + remove page scroll from demo

### PR DESCRIPTION
**Pull request by cristian-spiescu**
## Proposed Change:

Cf. #189 

## Change type

- [X] Bugfix
- [ ] New feature

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by bendog — APPROVED** [review]

**Comment by bendog** [comment]
seems to work for me, and it passes the tests.
@lilfolr any feedback?

File: src/components/body.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ class TimelineBody extends Component {
 TimelineBody.propTypes = {
   width: PropTypes.number.isRequired,
   columnWidth: PropTypes.func.isRequired,
-  height: PropTypes.number.isRequired,
+  height: PropTypes.oneOfType([PropTypes.number, PropTypes.func]).isRequired,
```
Comment (lilfolr) [inline src/components/body.js:54]: Is this required? Looks like we still pass in a number

File: src/timeline.js [hunk 2]
```diff
@@ -852,6 +852,14 @@ export default class Timeline extends React.Component {
         return width - groupOffset;
       };
     }
+
+    function calculateHeight(height) {
+      if (height === 0) return height;
+      // substract timebar height from total height
+      const timebarHeight = document.querySelector(`.rct9k-id-${componentId} .rct9k-timebar`).getBoundingClientRect().height;
+      return height - timebarHeight;
+    }
```
Comment (lilfolr) [inline src/timeline.js:861]: I doubt its possible `timebarHeight` is > `height`, but probably better to be safe:
```suggestion
    function calculateHeight(height) {
      // substract timebar height from total height
      const timebarHeight = document.querySelector(`.rct9k-id-${componentId} .rct9k-timebar`).getBoundingClientRect().height;
      return Math.max(height - timebarHeight, 0);
    }
```

**Comment by lilfolr** [comment]
If the above are addresses/justified I'm happy to merge

File: src/components/body.js (hunk 1 above)
Comment (eliza-marcu) [reply src/components/body.js:54]: We pass the function calculateHeight. I can remove number as required.

File: src/timeline.js (hunk 2 above)
Comment (eliza-marcu) [reply src/timeline.js:861]: It's a good point. I will commit this suggestion.

File: src/components/body.js (hunk 1 above)
Comment (lilfolr) [reply src/components/body.js:54]: Not sure If i'm missing something here...
The props is:
```javascript
height={calculateHeight(height)}
```
and `calculateHeight` returns a number, so the prop type is a number?

**Review by lilfolr — APPROVED** [review]

File: src/components/body.js (hunk 1 above)
Comment (eliza-marcu) [reply src/components/body.js:54]: You are right, I will remove ProtTypes.func as required.

**Review by lilfolr — APPROVED** [review]

**Comment by bendog** [comment]
Changes might be approved, but i'm not getting a functioning demo site anymore.
Screen Shot 2020-09-02 at 3 31 11 pm

**Review by bendog — CHANGES_REQUESTED** [review]
site no longer functioning

**Review by bendog — APPROVED** [review]

**Comment by bendog** [comment]
Thanks!

**Pull request merged**